include/mata/alphabet.hh

```diff
@@ -115,6 +117,8 @@ public:
 
     IntAlphabet& operator=(const IntAlphabet& int_alphabet) = delete;
 
+    void clear() override { throw std::runtime_error("Nonsensical use of clear() on IntAlphabet."); }
```

Comment:
Hmm, shouldn't this rather do nothing than throw error?

Reply:
I do not think so. When you run `clear()` on an instance of an `Alphabet*`, you obviously expect it to create an empty alphabet where you can insert symbols which will be the sole symbols in the alphabet. If the alphabet happens to be an `InAlpabet`, this invariant of `Alphabet` is invalid and theoretically you would have then check whether the alphabet instance is empty or something like that after running `clear()`. In my eyes, that is an anti-pattern to what you want when running `clear()`. But I am open to discussing this further.

Reply:
Ok, makes sense.